hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -129,10 +153,28 @@ public HandleContextImpl(
         this.serviceScopeLookup = requireNonNull(serviceScopeLookup, "serviceScopeLookup must not be null");
         this.blockRecordInfo = requireNonNull(blockRecordInfo, "blockRecordInfo must not be null");
         this.recordCache = requireNonNull(recordCache, "recordCache must not be null");
+        this.feeManager = requireNonNull(feeManager, "feeManager must not be null");
+        this.userTransactionConsensusTime =
+                requireNonNull(userTransactionConsensusTime, "userTransactionConsensusTime must not be null");
+        this.feeCalculatorCreator = (subType) -> feeManager.createFeeCalculator(
+                txInfo, payerKey, verifier.numSignaturesVerified(), userTransactionConsensusTime, subType);
```

Comment:
If we keep a reference to `txInfo` (instead of `txBody`) there is no need to define this lambda. We can call the method directly.

Reply:
The problem is we don't know the subtype, it isn't in the `txInfo` or the `txBody`, it is something that either we have to deduce from `txBody`, or something we have to get from the services themselves. I opted for the latter.